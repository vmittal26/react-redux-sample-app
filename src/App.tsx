import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import Header from './components/Header/Header';
import { Home, Login } from './pages';
import { AppState } from './shared';
import { connect, ConnectedProps } from 'react-redux';

import 'bootstrap/dist/css/bootstrap.min.css';

const mapStateToProps = ({ loginState }: AppState) => ({
  isAuthenticated: loginState.isAuthenticated
});

const connector = connect(mapStateToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

export const App = ({ isAuthenticated }: PropsFromRedux): React.ReactElement => {
  let routes = (
    <>
      <Header />
      <Switch>
        <Route exact path="/" component={Login} />
        <Redirect to="/" />
      </Switch>
    </>
  );

  if (isAuthenticated) {
    routes = (
      <>
        <Header />
        <Switch>
          <Route exact path="/" component={Login} />
          <Route exact path="/home" component={Home} />
          <Redirect to="/" />
        </Switch>
      </>
    );
  }
  return routes;
};

export default connector(App);
