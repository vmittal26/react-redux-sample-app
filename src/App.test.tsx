import React from 'react';
// import { render } from '@testing-library/react';
import { App } from './App';
import { shallow } from 'enzyme';

const mockDispatch = jest.fn();

test('renders learn app correctly if user is authenticated', () => {
  const wrapper = shallow(<App isAuthenticated dispatch={mockDispatch}/>);
  expect(wrapper).toMatchSnapshot();
});

test('renders learn app correctly if user is not authenticated', () => {
  const wrapper = shallow(<App isAuthenticated = {false} dispatch={mockDispatch}/>);
  expect(wrapper).toMatchSnapshot();
});
