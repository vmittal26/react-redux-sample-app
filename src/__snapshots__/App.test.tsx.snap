// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn app correctly if user is authenticated 1`] = `
<Fragment>
  <Connect(Header) />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/home"
    />
    <Redirect
      to="/"
    />
  </Switch>
</Fragment>
`;

exports[`renders learn app correctly if user is not authenticated 1`] = `
<Fragment>
  <Connect(Header) />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Redirect
      to="/"
    />
  </Switch>
</Fragment>
`;
